lib/posix/pthread.c

```diff
@@ -177,7 +176,7 @@ int pthread_create(pthread_t *newthread, const pthread_attr_t *_attr,
 	for (pthread_num = 0;
 	    pthread_num < CONFIG_MAX_PTHREAD_COUNT; pthread_num++) {
 		thread = &posix_thread_pool[pthread_num];
-		if (thread->state == PTHREAD_EXITED || thread->state == PTHREAD_TERMINATED) {
```

Comment:
yup, this seems like a good change. It does mean that you cannot re-use a thread until it has been joined again, right?

Reply:
Correct